### PR TITLE
Upload more than pdf

### PR DESCRIPTION
**Pull request by lemmsh**
Hi,

I need to be able to upload not only pdfs, but epub and fb2 as well. To make implementing this a bit easier, I've slightly refactored the web.py code, so someone else may add even more formats some time.

Thanks,
Pavel

### CODE REVIEW

**Comment by janeczku** [comment]
Hi Pavel @lemmsh, thanks for this PR and sorry for taking so long to respond. I will review this ASAP.

**Comment by janeczku** [comment]
@lemmsh sorry i am quite busy. Hope to find some time to review over the weekend. 💦

**Comment by lemmsh** [comment]
Hi,

That's fine, there is no rush :)

File: cps/book_formats.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    use_pdf_meta = True
+except ImportError, e:
+    logger.warning('cannot import PyPDF2, extracting pdf metadata will not work: %s', e)
+    use_pdf_meta = False
+
+try:
+    import epub
+    use_epub_meta = True
+except ImportError, e:
+    logger.warning('cannot import PyPDF2, extracting epub metadata will not work: %s', e)
```
Comment (cervinko) [inline cps/book_formats.py]: should be 'cannot import epub...' here

File: cps/book_formats.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    use_epub_meta = False
+
+try:
+    import fb2
+    use_fb2_meta = True
+except ImportError, e:
+    logger.warning('cannot import lxml, extracting fb2 metadata will not work: %s', e)
```
Comment (cervinko) [inline cps/book_formats.py]: cannot import fb2

File: cps/uploader.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+"""
+def upload(file):
+    if not os.path.isdir(tmp_dir):
+        os.mkdir(tmp_dir)
+
+    filename = file.filename
+    filename_root, file_extension = os.path.splitext(filename)
+    md5 = hashlib.md5()
+    md5.update(filename)
+    tmp_file_path = os.path.join(tmp_dir, md5.hexdigest())
```
Comment (cervinko) [inline cps/uploader.py]: would be better to use tempfile instead of md5sums

File: cps/fb2.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+from lxml import etree
+import os
+import uploader
+
+
+def get_fb2_info(tmp_file_path, original_file_name, original_file_extension):
+
+    ns = {
+        'fb':'http://www.gribuser.ru/xml/fictionbook/2.0',
+        'l':'ttp://www.w3.org/1999/xlink',
```
Comment (cervinko) [inline cps/fb2.py]: h in http missing

**Comment by cervinko** [comment]
i think we should include PyPDF2 and epub in vendor

**Pull request merged**